### ISSUE TITLE
gnplot 6.0.1 warning: Reading from '-' inside a multiplot not supported; use a datablock instead

### ISSUE THREAD

**Issue by daljit46** [alandefreitas/matplotplusplus#432]
In [Gnuplot 6.0.1](http://www.gnuplot.info/ReleaseNotes_6_0_1.html), the following change was introduced:

> CHANGE Use of data source '-' inside a multiplot is an error; use a local datablock instead 

This causes an excessive amount of warnings to be output to the console:

>  warning: Reading from '-' inside a multiplot not supported; use a datablock instead

I can easily reproduce this with the [line plot 1 example](https://github.com/alandefreitas/matplotplusplus/blob/master/examples/line_plot/plot/plot_1.cpp) on MacOS with gnuplot 6.0.1 installed via `brew`.

**Comment by Haron123** [comment]
Yeah i got the same issue, the amount of warnings printed is quite annoying

**Comment by SeverinDenisenko** [comment]
On my 1024x1024 image plot it logs over 30000 warnings per second, rendering application completely unusable.

For a workaround I inserted `run_command("unset warnings");` in some place in code.

This is obviously not a fix and I don't know what I am doing. But this worked.

**Comment by shen13203708051** [comment]
Hello, I have also encountered this problem, and it seems to be easy to occur, even with the official code. 
`#include <cmath>
#include <matplot/matplot.h>

int main() {
    using namespace matplot;
    std::vector<double> x = linspace(0, 2 * pi);
    std::vector<double> y = transform(x, [](auto x) { return sin(x); });

    plot(x, y, "-o");
    hold(on);
    plot(x, transform(y, [](auto y) { return -y; }), "--xr");
    plot(x, transform(x, [](auto x) { return x / pi - 1.; }), "-:gs");
    plot({1.0, 0.7, 0.4, 0.0, -0.4, -0.7, -1}, "k");

    show();
    return 0;
}`
I am using the latest version currently available, gunplot 6.0.2, windows C++ 17, matplot++ 5d01eb3. How did you solve this problem later? Can you share? And I didn't know how to operate the `run_command` in the previous answer,  but it seems to have worked. @daljit46 @SeverinDenisenko @Haron123

**Comment by Xyhlon** [comment]
> Hello, I have also encountered this problem, and it seems to be easy to occur, even with the official code.
> 
> ```
> #include <cmath>
> #include <matplot/matplot.h>
> 
> int main() {
>     using namespace matplot;
>     std::vector<double> x = linspace(0, 2 * pi);
>     std::vector<double> y = transform(x, [](auto x) { return sin(x); });
> 
>     plot(x, y, "-o");
>     hold(on);
>     plot(x, transform(y, [](auto y) { return -y; }), "--xr");
>     plot(x, transform(x, [](auto x) { return x / pi - 1.; }), "-:gs");
>     plot({1.0, 0.7, 0.4, 0.0, -0.4, -0.7, -1}, "k");
> 
>     show();
>     return 0;
> }
> ```
> 
> I am using the latest version currently available, gunplot 6.0.2, windows C++ 17, matplot++ [5d01eb3](https://github.com/alandefreitas/matplotplusplus/commit/5d01eb3695b07634a2b6642fd423740dea9b026c). How did you solve this problem later? Can you share? And I didn't know how to operate the `run_command` in the previous answer, but it seems to have worked. [@daljit46](https://github.com/daljit46) [@SeverinDenisenko](https://github.com/SeverinDenisenko) [@Haron123](https://github.com/Haron123)

like so:
```cpp
    auto f = figure(false);
    f->backend()->run_command("unset warnings");
    f->ioff();
    f->size(1600, 900);
    ...
```
